src/adapters/yieldbot.js

```diff
@@ -83,24 +83,35 @@ var YieldbotAdapter = function YieldbotAdapter() {
 
       ybotq.push(function () {
         var yieldbot = window.yieldbot;
-
+        // Empty defined slots bidId array
         ybotlib.definedSlots = [];
+        // Iterate through bids to obtain Yieldbot slot config
+        // - Slot config can be different between initial and refresh requests
+        var psn = 'ERROR_PREBID_DEFINE_YB_PSN';
+        var slots = {};
         utils._each(bids, function (v) {
           var bid = v;
-          var psn = bid.params && bid.params.psn || 'ERROR_DEFINE_YB_PSN';
-          var slot = bid.params && bid.params.slot || 'ERROR_DEFINE_YB_SLOT';
+          // bidder params config: http://prebid.org/dev-docs/bidders/yieldbot.html
+          // - last psn wins
+          psn = bid.params && bid.params.psn || psn;
+          var slotName = bid.params && bid.params.slot || 'ERROR_PREBID_DEFINE_YB_SLOT';
 
-          yieldbot.pub(psn);
-          yieldbot.defineSlot(slot, { sizes: bid.sizes || [] });
+          slots[slotName] = bid.sizes || [];
           ybotlib.definedSlots.push(bid.bidId);
         });
-        yieldbot.enableAsync();
+
         if (yieldbot._initialized !== true) {
+          yieldbot.pub(psn);
+          for (var slotName in slots) {
```

Comment:
@grevory added the check for own property and a test for the check.

The object being checked is created `var slots = {};` so, I expect pretty unlikely to be exposed to any injected properties causing an issue with iteration of the enumerable properties... However, it's _been known_ that overriding Object in the wild is not implausible.

Please comment here if you were thinking of any specific issues you had in mind.

Understood that it's generally accepted good form to be defensive on inherited properties.

If you're ok with this, I can squash the fork and merge the fork with upstream master. lmk.

Thanks!